src/shared/hooks/useSavedCourses.ts

```diff
@@ -7,41 +13,141 @@ export type Course = {
   code: string;
   pid: string;
   term: string;
+  sections: Section[];
+  selected?: boolean;
+  color?: string;
+  textColor?: string;
+  lecture?: SavedSection;
+  lab?: SavedSection;
+  tutorial?: SavedSection;
+};
+
+export type SavedSection = {
+  meetingTimes: MeetingTimes[];
+  sectionCode: string;
 };
 
 type SavedCourses = {
   courses: Course[];
   addCourse: (newCourse: Course) => void;
   deleteCourse: (newCourse: Course) => void;
   clearCourses: () => void;
+  setSection: (type: string, newSection: SavedSection, existingCourse: Course) => void;
   contains: (pid: string, term: string) => boolean;
+  sectionIsSaved: (pid: string, term: string, sectionCode: string) => boolean;
+  setSelected: (selected: boolean, existingCourse: Course) => void;
```

Comment:
This could be a bit more clear. At first glance not sure what it does 100% and what needs to be passed in.